tests/simulator/test_advance.py

```diff
@@ -0,0 +1,240 @@
+        datahier = self.getHierarchy(interp_order, refinement_boxes, "eb", cells=30,
+                                      diag_outputs=diag_outputs,
+                                      extra_diag_options={"fine_dump_lvl_max": 10},
+                                      advances=advances, smallest_patch_size=5, largest_patch_size=30)
+
+        times = datahier.times()
+        for step in range(1, totalSteps + 1):
+            checkTime = float("{:.6f}".format(finestTimeStep * step))
+            self.assertIn(checkTime, times)
+            uniqTimes.add(checkTime)
+
+        self.assertEqual(len(uniqTimes), len(datahier.time_hier.items()))
+
+        syncSteps = nSubcycles ** levelNumbers[-2] * advances # ignore most fine subcycles
+
+        # FIX THIS AFTER NO MORE REGRIDS
+        #  SEE: https://github.com/PHAREHUB/PHARE/issues/400
+        assert syncSteps % advances == 0 # perfect division
+        startStep = int(syncSteps / advances) + 1 # skip coarsest step due to issue 400
+        for step in range(startStep, syncSteps + 1):
+            checkTime = float("{:.6f}".format(secondFinestTimeStep * step))
+            self.assertIn(checkTime, datahier.times())
+            nLevels = len(datahier.time_hier[checkTime].items())
+            self.assertGreaterEqual(nLevels, 2)
+            levelNbrs = list(datahier.time_hier[checkTime].keys())
+            finestLevelNbr = max(levelNbrs)
+            coarsestLevelNbr = min(levelNbrs)
+
+            for coarseLevelNbr in range(coarsestLevelNbr, finestLevelNbr):
+                coarsePatches = datahier.time_hier[checkTime][coarseLevelNbr].patches
+                finePatches = datahier.time_hier[checkTime][coarseLevelNbr + 1].patches
+
+                for coarsePatch in coarsePatches:
+                    for finePatch in finePatches:
+                        lvlOverlap = boxm.refine(coarsePatch.box, 2) * finePatch.box
+                        if lvlOverlap is not None:
+                            for EM in ["E", "B"]:
+                                for xyz in ["x", "y", "z"]:
+                                    qty = f"EM_{EM}_{xyz}"
+                                    coarse_pd = coarsePatch.patch_datas[qty]
+                                    fine_pd  = finePatch.patch_datas[qty]
+                                    coarseBox = boxm.coarsen(lvlOverlap, 2)
+
+                                    qty = f"{EM}{xyz}"
+                                    nGhosts = coarse_pd.layout.nbrGhostFor(qty)
+
+                                    coarse_pdDataset = coarse_pd.dataset[:]
+                                    fine_pdDataset = fine_pd.dataset[:]
+
+                                    coarseOffset = coarseBox.lower - coarse_pd.layout.box.lower
+                                    dataBox_lower = coarseOffset + nGhosts
+                                    dataBox = Box(dataBox_lower, dataBox_lower + coarseBox.shape() - 1)
```

Comment:
the PatchData has a box already from its layout, could that be of use here?

Reply:
this is the area within the coarse patch_data that is expected to receive coarsened values, it's already using the box on the layout.

Reply:
I misread, my bad